From: Alberto Garcia <berto@igalia.com>
Bug: https://bugs.webkit.org/show_bug.cgi?id=172799
Description: Use CMAKE_CXX_LIBRARY_ARCHITECTURE to detect x86
 When building for x86 in a x86_64 host (e.g. with pbuilder) CMake
 will set CMAKE_SYSTEM_PROCESSOR to x86_64, and WebKit will assume
 that it's a 64-bit build. Let's use the C++ compiler architecture as
 a temporary solution to detect x86 correctly.
Index: webkitgtk/CMakeLists.txt
===================================================================
--- webkitgtk.orig/CMakeLists.txt
+++ webkitgtk/CMakeLists.txt
@@ -75,6 +75,10 @@ elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR
     set(WTF_CPU_MIPS 1)
 elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR MATCHES "sh4")
     set(WTF_CPU_SH4 1)
+elseif (CMAKE_CXX_LIBRARY_ARCHITECTURE MATCHES "(x64|x86_64|amd64)")
+    set(WTF_CPU_X86_64 1)
+elseif (CMAKE_CXX_LIBRARY_ARCHITECTURE MATCHES "(i[3-6]86|x86)")
+    set(WTF_CPU_X86 1)
 elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR MATCHES "(x64|x86_64|amd64)")
     set(WTF_CPU_X86_64 1)
 elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR MATCHES "(i[3-6]86|x86)")
